src/App.tsx

```diff
@@ -1,24 +1,27 @@
-import React from 'react';
-import logo from './logo.svg';
 import './App.css';
 
+import { useState } from 'react';
+import TodoList from './Components/ToDoList';
+import { ToDoInterface } from './Shared/interface';
+import { sampleDataForList } from './Shared/sample';
+
 function App() {
+  const toDoArray:ToDoInterface[] = sampleDataForList;
+  const [todos, setTodos] = useState(toDoArray);
+
+  const markCompleted = (id:number) => {
+    setTodos((prevTodos:ToDoInterface[]) =>
+      prevTodos.map((todo:ToDoInterface) => (todo.id === id ? { ...todo, completed: !todo.completed } : todo))
+    );
+  };
+
+  const deleteTodo = (id:number) => {
+    setTodos((prevTodos:ToDoInterface[]) => prevTodos.filter((todo) => todo.id !== id));
```

Comment:
is previous state required here? I mean (prevTodos:ToDoInterface[])  is this required?

Reply:
Yes

Here we need to maintain previous state while  updating the data

Reply:
can you try with 
setTodos(( todos.filter((todo) => todo.id !== id));

Reply:
Yes it works as expected.

I willl note this point